Editor: Wrap long debugger error messages in tooltips

isrichards6 opened the pull request:
Implements text wrapping for debugger tooltips using TextServer, fixing readability issues for long error messages. Uses TextServer::string_get_word_breaks() with a character limit to insert newlines into the tooltip string generated in _msg_error.

Fixes #105634 

Tested using the MRP provided in the issue and verified that long error messages now wrap correctly within the tooltip bounds.

Before:
![before](https://github.com/user-attachments/assets/fa6c30bd-f50a-4e12-bc93-fa586253da44)

After:
![after](https://github.com/user-attachments/assets/4fab18a0-78f4-4a22-8eb7-27a486fec408)

isrichards6 committed:
Editor: Wrap long debugger error messages in tooltips

Implements text wrapping for debugger tooltips using TextServer, fixing readability issues for long error messages. Uses TextServer::string_get_word_breaks() with a character limit to insert newlines into the tooltip string generated in _msg_error.

Vulwsztyn commented:
IMO readability of the code would benefit if this functionality was extracted into a function.

I also feel like we could get rid of the comments in this PR if we introduced more splitting into functions or descriptive variable names.

AThousandShips commented:
I don't agree, it would separate the use from the implementation, since this is used in one place only

iprichards commented:
> IMO readability of the code would benefit if this functionality was extracted into a function.
> 
> I also feel like we could get rid of the comments in this PR if we introduced more splitting into functions or descriptive variable names.

I tried to keep this fix in-line with number  **5: To each problem, its own solution** from [Best practices for engine contributors](https://docs.godotengine.org/en/stable/contributing/development/best_practices_for_engine_contributors.html?hl=en-US#to-each-problem-its-own-solution).

With the reasoning being there currently is not a need for multiple instances of this fix being needed where a function would prove most useful.

Vulwsztyn commented:
Hmm. Both of you use the reasoning that challanges my views. I do not look at functions as something that should only exist if there is a nerd to use the code multiple times. Imo functions serve for dividing code into logically separate chunks, and layers of abstraction separation, which allows the functions that use them to be shorter and thus more parable for a human. This view might be in contradiction to the philosophy here, and as I'm not a maintainer, nor even a contributor yet, my opinion should not matter in the final decision.

isrichards6 committed:
Merge branch 'master' into fix-debugger-tooltip-wrap

AThousandShips commented:
Not sure why this is but it looks like this PR is empty, no changed files

Is this PR still relevant?

isrichards6 commented:
@AThousandShips Currently checking on this. I believe the original issue I addressed (#105634) got fixed since I originally pushed this, which is why I removed my code. Will close this PR and post an update on the issue page if I find that to be the case, otherwise I will look into reimplementing a fix.

AThousandShips commented:
If you removed the code I'd say to close this PR as it is no longer valid, and if still relevant to open a new PR (not sure why this PR wasn't automatically closed it should have been)

isrichards6 commented:
Verified that fix is still needed, closing this PR and will open new one.
